wagtail/admin/views/pages/delete.py

```diff
@@ -24,17 +25,44 @@ def delete(request, page_id):
 
         next_url = get_valid_next_url_from_request(request)
 
+        pages_to_delete = {page}
+
+        # The `construct_synced_page_tree_list` hook returns translation and
+        # alias pages when the action is set to "delete"
+        if getattr(settings, "WAGTAIL_I18N_ENABLED", False):
+            for fn in hooks.get_hooks("construct_synced_page_tree_list"):
+                fn_pages = fn([page], "delete")
+                if fn_pages and isinstance(fn_pages, dict):
+                    for additional_pages in fn_pages.values():
+                        pages_to_delete.update(additional_pages)
+
+        pages_to_delete = list(pages_to_delete)
+
         if request.method == "POST":
             parent_id = page.get_parent().id
+            # Delete the source page.
             action = DeletePageAction(page, user=request.user)
             # Permission checks are done above, so skip them in execute.
             action.execute(skip_permission_checks=True)
 
+            # Delete translation and alias pages if they have the same parent page.
+            if getattr(settings, "WAGTAIL_I18N_ENABLED", False):
+                parent_page_translations = page.get_parent().get_translations()
+                for _page in pages_to_delete:
+                    if _page.get_parent() in parent_page_translations:
+                        action = DeletePageAction(_page, user=request.user)
+                        # Permission checks are done above, so skip them in execute.
+                        action.execute(skip_permission_checks=True)
+
             messages.success(
                 request, _("Page '{0}' deleted.").format(page.get_admin_display_title())
             )
 
             for fn in hooks.get_hooks("after_delete_page"):
+                pages_to_delete.remove(page)
+                for _page in pages_to_delete:
+                    fn(request, _page)
```

Comment:
Going to backtrack on this a bit. Sorry @KalobTaulien 

as per https://docs.wagtail.org/en/stable/reference/hooks.html these hooks only apply for the page on which the action is performed via the UI. in this case that is the source page. So this bit is extra 🤦‍♂️😂